### DISCUSSION TITLE
Unable to run ao3downloader

### DISCUSSION THREAD

**Discussion by j-p-garcia** [nianeyna/ao3downloader#56] [Q&A] [answered]
I downloaded the updated version of ao3downloader and now am unable to run the code, with this error message:

'venv\Scripts\Activate' is not recognized as an internal or external command, operable program or batch file.
Traceback (most recent call last):  File "C:\Users\kuros\Downloads\ao3downloader-main(1)\ao3downloader-main\ao3downloader.py", line 3, in <module>
    from ao3downloader.actions import ao3download
  File "C:\Users\kuros\Downloads\ao3downloader-main(1)\ao3downloader-main\ao3downloader\actions\ao3download.py", line 1, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
Press any key to continue . . .

**Comment by nianeyna** [comment]
Interesting, you're the second person to report this in the last few days. Are you on Windows 11 by any chance?

**Reply by j-p-garcia** [reply]
I'm not, actually, I'm still on Windows 10.

**Reply by nianeyna** [reply]
Darn. I think that rules out my theory that it was accidentally being opened in Windows Terminal. I'll keep looking into it - I haven't been able to reproduce this one unfortunately so it's tough to debug. I'm fairly sure it's not caused by any of the code changes I've made recently, since the problem appears to be entirely in the startup script which I haven't touched since March.

**Comment by nianeyna** [comment]
Try replacing the contents of ao3downloader.cmd (right-click -> Edit) with the following:

```
@echo off

if not exist "env\" (set firstrun="true") else (set firstrun="false")

if %firstrun%=="true" python -m venv env

call env\Scripts\activate

if %firstrun%=="true" python -m pip install --upgrade pip & pip install -r requirements.txt

python ao3downloader.py

pause
```
Let me know how it goes.

**Reply by j-p-garcia** [reply]
No change, except that the error message now said "env" instead of "venv"

**Reply by nianeyna** [reply]
Rats. I'll keep looking. Does `env\Scripts\activate` exist? That is, is there a folder called "env" which contains a folder called "Scripts" which contains a file called "activate"?

**Reply by j-p-garcia** [reply]
Both "env" and "venv" exist, and they each have a folder called "Scripts". Both "Scripts" files contain two files, both of which appear to be the application Python?

**Reply by nianeyna** [reply]
Hm. There should definitely be some files named "activate" in there too. Plus quite a few other things. So for some reason your virtual environment is coming out half-baked. Dunno what could be causing that. Can you please open a command prompt and tell me the output of the commands `where python` and `where pip`?

**Reply by j-p-garcia** [reply]
They're both in AppData\Local\Microsoft\WindowsApps.

I have used the older version of the downloader before, and when I checked the venv\Scripts in that older file it did have "activate" as both a windows batch file and a windows powershell script. I have no idea if that information is useful but yeah

**Comment by nianeyna** [comment]
Please follow the instructions here: https://stackoverflow.com/questions/58754860/cmd-opens-windows-store-when-i-type-python/58773979#58773979

Then take a fresh copy of the script from github (don't use the one you've already downloaded), and try running the script again.

**Reply by j-p-garcia** [reply]
i have done something truly baffling to my computer which is absolutely not your problem. thank you so much for your patience with my (tech and general) illiteracy and i'll presumably get this figured out eventually

**Reply by nianeyna** [reply]
I'm sorry I couldn't help! If you do find a solution I'd be interested to hear what the problem wound up being. Surprised this one didn't work, I was sure that had to be it.

**Comment by gildobi** [comment]
Hello, I had the same problem, but I used python 3.11.0 (Dec 7, 2022). So I uninstalled the python and deleted the unzip folder ao3downloader-main. I downloaded python 3.9.0 (released before the date you posted this problem - Sep 17), installed it, unzipped the file, and ran smoothly. I think you should try this.

**Reply by soph104** [reply, marked as answer]
Thank you so much, that finally fixed it for me. I had tried 3.9.9 and the current release but in 3.9.0 it works finally
